mtags/src/main/scala-2/scala/meta/internal/pc/MetalsGlobal.scala

```diff
@@ -306,238 +306,245 @@ class MetalsGlobal(
-            // [x] => F[x] is not printable in the code, we need to use just `F`
-            case TypeRef(_, sym, args)
-                if typeParams == args.map(_.typeSymbol) =>
-              TypeRef(
-                NoPrefix,
-                sym.newErrorSymbol(sym.name),
-                Nil
-              )
-            case otherType =>
-              PolyType(typeParams, otherType)
-          }
-        case NullaryMethodType(resultType) =>
-          loop(resultType, None)
-        case TypeBounds(lo, hi) =>
-          TypeBounds(loop(lo, None), loop(hi, None))
-        case MethodType(params, resultType) =>
-          MethodType(params, loop(resultType, None))
-        case ErrorType =>
-          definitions.AnyTpe
-        case t => t
+          case ConstantType(Constant(sym: TermSymbol))
+              if sym.hasFlag(gf.JAVA_ENUM) =>
+            loop(SingleType(sym.owner.thisPrefix, sym), None)
+          case ConstantType(Constant(tpe: Type)) =>
+            ConstantType(Constant(loop(tpe, None)))
+          case SuperType(thistpe, supertpe) =>
+            SuperType(loop(thistpe, None), loop(supertpe, None))
+          case RefinedType(parents, decls) =>
+            RefinedType(parents.map(parent => loop(parent, None)), decls)
+          case AnnotatedType(annotations, underlying) =>
+            AnnotatedType(annotations, loop(underlying, None))
+          case ExistentialType(quantified, underlying) =>
+            ExistentialType(
+              quantified.map(sym => sym.setInfo(loop(sym.info, None))),
+              loop(underlying, None)
+            )
+          case PolyType(typeParams, resultType) =>
+            resultType.map(t => loop(t, None)) match {
+              // [x] => F[x] is not printable in the code, we need to use just `F`
+              case TypeRef(_, sym, args)
+                  if typeParams == args.map(_.typeSymbol) =>
+                TypeRef(
+                  NoPrefix,
+                  sym.newErrorSymbol(sym.name),
+                  Nil
+                )
+              case otherType =>
+                PolyType(typeParams, otherType)
+            }
+          case NullaryMethodType(resultType) =>
+            loop(resultType, None)
+          case TypeBounds(lo, hi) =>
+            TypeBounds(loop(lo, None), loop(hi, None))
+          case MethodType(params, resultType) =>
+            MethodType(params, loop(resultType, None))
+          case ErrorType =>
+            definitions.AnyTpe
+          case t => t
+        }
+        cached.putIfAbsent(key, result)
+        result
       }
-      cached.putIfAbsent(key, result)
-      result
-    }
 
-    longType match {
-      case ThisType(_) => longType
-      case _ => loop(longType, None)
+      longType match {
+        case ThisType(_) => longType
+        case _ => loop(longType, None)
+      }
+    } catch {
+      case NonFatal(e) =>
+        logger.warning(s"Failed to shorten type $longType: $e")
```

Comment:
maybe wrap the exception with that info instead? that way, the stack trace will include both this message and everything else it normally does. Maybe even include the `history` parameter in there if that makes sense

Reply:
> maybe wrap the exception with that info instead?

Are suggesting to add the info and rethrow?

One thing that I don't understand is how that error even surfaced like this. Errors inside of pc are caught and reported as error reports. What version of Scala was that exactly?

Reply:
> Are suggesting to add the info and rethrow?

yep

> What version of Scala was that exactly?

I don't even remember, but it could even have been a cross-compiled project with sbt-projectmatrix (meaning all build variants would have their bloop project created, and loaded into Metals) 😬 

next time I hit this (or something similar), I'll make sure to gather more info.

Reply:
Makes sense, thanks